source/widgets/UnixWindow.ooc

```diff
@@ -43,5 +44,15 @@ UnixWindow: class extends DisplayWindow {
 	refresh: override func {
 		this _openGLWindow refresh()
 	}
+	processEvents: override func {
+		while (XPending(this _xWindow display)) {
+			event: XEventOoc
+			XNextEvent(this _xWindow display, event&)
+			if ((event type == ButtonPress) && this _mousePressHandler)
+				this _mousePressHandler call(IntPoint2D new(event xkey x, event xkey y))
+			if ((event type == ButtonRelease) && this _mouseReleaseHandler)
+				this _mouseReleaseHandler call(IntPoint2D new(event xkey x, event xkey y))
+		}
+	}
```

Comment:
Could be done that way, but I will have to move the mouse event handlers there as well and create functions to forward then into the xWindow from DisplayWindow. This will add some work later when I will clean this up anyway.